### ISSUE TITLE
Sniffer rules add PHP 5.6 requirement

### ISSUE THREAD

**Issue by sdustinh** [cakephp/cakephp-codesniffer#118]
So this is kind of a weird one. I extended the FlashHelper to work with Bootstrap. When I did I went ahead and added some templates to help with generation. This is what I ended up with...

```php
protected $_defaultConfig = [
    'linkClass' => 'alert-link',
    'templates' => [
        'dismiss' => '<button type="button" class="close" data-dismiss="alert" aria-label="{{label}}"{{attrs}}>{{content}}</button>',
        'dismissIcon' => '<span aria-hidden="true">&times;</span>'
    ]
];
```

This causes a phpcs error as the `dismiss` key has 134 characters on a single line. So I converted it to this...

```php
protected $_defaultConfig = [
    'linkClass' => 'alert-link',
    'templates' => [
        'dismiss' => 
            '<button type="button" class="close" data-dismiss="alert" aria-label="{{label}}"{{attrs}}>' .
                '{{content}}' .
            '</button>',
        'dismissIcon' => '<span aria-hidden="true">&times;</span>'
    ]
];
```

This passes the sniffer rules so I pushed to the repo all well and good. I went to make some changes on my computer at home and I ended up with a fatal PHP error saying it expected a `]` character. When I collapse everything back onto a single line the error goes away.

I'm not sure this is anything you guys would be responsible for in terms of updating your sniffer rules but it seems like such an obscure error that I just wanted to report it so you were aware.

I also seem to get conflicts when running an underscored method name such as...

```php
protected function _getSomething()
{
}
```

With the underscore I get a PSR2 error. Without the underscore I get a CakePHP error. Weird. Here is the version I'm running at home. I'm not in the office so I don't have the exact output for that one but it's a fresh install of php 5.6.4.

```
dustin@Dustin:~/GitHub/internal$ php --version
PHP 5.5.20-1+deb.sury.org~precise+1 (cli) (built: Dec 21 2014 19:54:33) 
Copyright (c) 1997-2014 The PHP Group
Zend Engine v2.5.0, Copyright (c) 1998-2014 Zend Technologies
    with Zend OPcache v7.0.4-dev, Copyright (c) 1999-2014, by Zend Technologies
    with Xdebug v2.2.5, Copyright (c) 2002-2014, by Derick Rethans
```

**Comment by jadb** [comment]
@sdustinh for the protected method - turn warnings off (use `phpcs -n`). CakePHP doesn't fully follow PSR2.

As for the long line error, try defining your `$_defaultConfig` in the `__construct()` - I don't think you can to [string concatenation when defining class properties](http://stackoverflow.com/questions/8631653/php-string-concatenation-within-class-definition) (maybe on php 5.6 which you seem to have at work but I haven't tried that).

**Comment by sdustinh** [comment]
@jadb Ah, you're right. You can't do it in [php 5.5 but you can in 5.6](http://docs.php.net/manual/en/migration56.new-features.php#migration56.new-features.const-scalar-exprs). I guess I could throw it into the constructor but it just doesn't look at uniform. :)

I guess I should have looked into this a little deeper before opening the issue. Sorry about that, thanks for the help!
